### DISCUSSION TITLE
Create generic gpio in device tree.

### DISCUSSION THREAD

**Discussion by alexanderhoet** [zephyrproject-rtos/zephyr#33307] [Q&A] [answered]
Hello,

I have recently decided to give Zephyr a try and thus far I am really liking what I see, also I really like the divice tree approach. I've been able to get Leds, SPI, SPI Flash and PWM working in no time (ok, doing it the old-fashioned way would have been faster but we're on a learning curve).
But just now I ran into the following problem:

How do I create a generic gpio in the device tree?

I know how to create leds and keys, but this one is neither. I have tried adding the following to the root node:

`
clk_en: clk_en {
	gpios = <&gpioh 1 GPIO_ACTIVE_HIGH>;
	label = "Clock Enable";
};
`

and the node shows up, it 'exists' and the label is created. But the gpios property is not.
For now I have solved it by putting it with my leds, but this just seems dirty as this output clearly isn't an LED.

**Comment by drojf** [comment]
A similar question was asked here: https://github.com/zephyrproject-rtos/zephyr/discussions/29281

...I think the answer was to just use `gpio-leds`, or the inferred-bindings, but that's probably not the answer you're looking for

**Comment by henrikbrixandersen** [comment, marked as answer]
As @drojf wrote, there are multiple options. Either use the inferred bindings (see https://docs.zephyrproject.org/latest/guides/dts/bindings.html#inferred-bindings) or create a custom, board specific devicetree binding (see https://docs.zephyrproject.org/latest/application/index.html#custom-board-devicetree-and-soc-definitions for how to add it, https://docs.zephyrproject.org/latest/guides/dts/bindings.html for bindings syntax, and https://docs.zephyrproject.org/latest/guides/dts/intro.html#devicetree-intro for an intro on the bindings are used).

The Arty board definition uses a board specific DTS binding containing GPIO pin(s). You might find some inspiration there: https://github.com/zephyrproject-rtos/zephyr/tree/master/boards/arm/arty/dts/bindings
